src/library/scala/concurrent/Future.scala

```diff
@@ -101,9 +107,13 @@ trait Future[+T] extends Awaitable[T] {
    *  If the future has already been completed with a value,
    *  this will either be applied immediately or be scheduled asynchronously.
    *
+   *  Note that the returned value of `pf` will be discarded.
+   *
    *  $swallowsExceptions
    *  $multipleCallbacks
    *  $callbackInContext
+   *
+   * @group Callbacks
```

Comment:
Looks like lexicographical so with `Transformations` not sandwiched by `Polling` and `Callbacks` the default ordering is fine.

![scaladoc-futures](https://cloud.githubusercontent.com/assets/1123855/15293574/8624dfe2-1b80-11e6-8db4-637225ca09ba.png)

Is `andThen` more of a transformation than a callback?